src/openms_ml/include/OpenMS/ML/AlphaDatahandling.h

```diff
@@ -33,20 +33,25 @@
 // --------------------------------------------------------------------------
 
 #pragma once
-
-#include <OpenMS/OpenMS_MLConfig.h>
-
-#include <vector>
+#include <memory> 
 #include <string>
+#include <vector>
 
-// Note: No torch headers may be included here! only in c++ file. otherwise torch will pollute whole OpenMS API
 namespace OpenMS
 {
-	class OPENMS_ML_DLLAPI DeepLCWrapper
-	{
-		explicit DeepLCWrapper(const std::string& filename);
-
-		std::vector<double> predict(const std::vector<std::string>& sequences, const std::vector<double>& obs_rt); // TODO: other inputs... encoding etc.
-	};
-
-}
+  class AlphaDatahandler
```

Comment:
I am not sure if I understand the intent of this class correctly – maybe because of the missing documentation.

In its current state it is not really a class that creates and stores the features, as it stores also the original data that is passed (is this needed?).
But it doesn't seem to be a complete representation of the feature vectors/matrix as it still needs to provide some functions like get_batch_aa_indices.

This looks, at first glance, like the class is doing two different things. Maybe there are good reasons for it but it is not clear from looking at this class alone and it could be an indicator that there is something fishy with the design.

One other downside with your approach could be that this class stores a lot of data internally.
I wonder if one could get rid of storing this internal data without performance impact.
E.g., what would be the impact of 
turning:
```std::vector<std::vector<long int>> get_batch_aa_indices();```
into:
```std::vector<std::vector<long int>> get_batch_aa_indices(const std::vector<std::string>& seq);```
so that no internal data needs to be stored?
and same for get_batch_mod_features.

E.g. I would assume one just needs a
static std::vector<std::vector<double>> getFeatureMatrix(const std::vector<std::string>& seq)
or
static std::map<std::string, std::vector<double>> getFeatureMatrix(const std::vector<std::string>& seq)
where the map key is the feature column name.

we can also zoom to discuss this a bit more in detail if you want

Reply:
e.g. similar how we do in https://github.com/OpenMS/OpenMS/blob/develop/src/openms/include/OpenMS/ANALYSIS/SVM/SimpleSVM.h#L81

Reply:
> One other downside with your approach could be that this class stores a lot of data internally.
I wonder if one could get rid of storing this internal data without performance impact.
E.g., what would be the impact of
turning:
std::vector<std::vector<long int>> get_batch_aa_indices();
into:
std::vector<std::vector<long int>> get_batch_aa_indices(const std::vector<std::string>& seq);
so that no internal data needs to be stored?
and same for get_batch_mod_features.

>> Good point can possible like that. so data will not stored, instead just use in function.